core/include/usFrameworkEvent.h

```diff
@@ -56,7 +56,10 @@ class US_Core_EXPORT FrameworkEvent
 
 public:
 
-  enum Type {
+  /**
+   * A type code used to identify the event type for future extendability.
+   */
+  enum class Type : unsigned int {
```

Comment:
The other event classes use an unscoped enum with a `uint32_t` type. We should be consistent.